pkg/packet/bgp/bgp.go

```diff
@@ -12440,6 +12517,11 @@ func (msg *BGPUpdate) DecodeFromBytes(data []byte, options ...*MarshallingOption
 	if len(data) < int(msg.TotalPathAttributeLen) {
 		return NewMessageError(eCode, eSubCode, nil, "path total attribute length exceeds message length")
 	}
+	attributes := GetBGPUpdateAttributes(data)
+	o := MarshallingOption{
+		Attributes: &attributes,
+	}
+	options = append(options, &o)
```

Comment:
Can we skip this for non AFI 1 SAFI 128 case for the performance?

Reply:
It is doable but I am not sure it will be more performant, with this approach all we check is for a presence of an attribute and then set bool flag in the map. If you want to be selective, then I would need to look deeper, first finding mp_reach_nlri in the list of attributes and then get afi/safi.  We do not see more than 5-8 attributes in the update, so it should be fairly quick. Please let me know if you still want me to do it only for afi/safi 1/128.

Reply:
Ok, leave it alone. We could improve this later anyway.